README.md

```diff
@@ -27,14 +46,33 @@ make docs
 ```
 The documentation can then be found at `docs/html/index.html`
 
+## Tests
+To run unit tests and jlc C compilation tests, execute
+```
+make check
+```
+
+The tests can also be run instrumented under valgrind to validate absence
+of detectable memory errors:
+```
+make valgrind-check
+```
+
+Lastly, when the build has been configured with coverage support, then
+the following build target will compute unit test coverage for all files:
+```
+make coverage
```

Comment:
The code coverage creates a dependency to `gcovr`. It should be added to the dependency list in the README.md.

Reply:
added explanation to README.md